pfcli/service/client/data.py

```diff
@@ -65,18 +79,255 @@ def delete_datastore(self, datastore_id: T) -> None:
+                "upload_urls": [
+                    {
+                        "upload_url": "string",
+                        "part_number": 0
+                    }
+                ]
+            }
+        """
+        start_mpu_resps = []
+        for path in paths:
+            if src_path is not None and not os.path.isfile(path):
+                num_parts = math.ceil(get_file_size(os.path.join(src_path, path)) / S3_MPU_PART_MAX_SIZE)
+            else:
+                num_parts = math.ceil(get_file_size(path) / S3_MPU_PART_MAX_SIZE)
+            response = safe_request(self.post, err_prefix="Failed to get presigned URLs for multipart upload.")(
+                path=f"{datastore_id}/start_mpu/",
+                json={
+                    "path": path,
+                    "num_parts": num_parts,
+                }
+            )
+            start_mpu_resps.append(response.json())
+        return start_mpu_resps
+
+    def complete_mpu(self, datastore_id: T, path: str, upload_id: str, parts: List[dict]) -> None:
+        safe_request(self.post, err_prefix=f"Failed to complete multipart upload for {path}")(
+            path=f"{datastore_id}/complete_mpu/",
+            json={
+                "path": path,
+                "upload_id": upload_id,
+                "parts": parts,
+            }
+        )
+
+    def abort_mpu(self, datastore_id: T, path: str, upload_id: str) -> None:
+        safe_request(self.post, err_prefix=f"Failed to abort multipart upload for {path}")(
+            path=f"{datastore_id}/abort_mpu/",
+            json={
+                "path": path,
+                "upload_id": upload_id,
+            }
+        )
+
+    def _multipart_upload_file(
+        self,
+        datastore_id: T,
+        file_path: str,
+        url_dict: List[dict],
+        ctx: tqdm
+    ) -> None:
+        parts = []
+        upload_id = url_dict['upload_id']
+        object_path = url_dict['path']
+        try:
+            with open(file_path, 'rb') as f:
+                fileno = f.fileno()
+                total_file_size = os.fstat(fileno).st_size
+                for idx, url_info in enumerate(url_dict['upload_urls']):
+                    cursor = idx * S3_MPU_PART_MAX_SIZE
+                    f.seek(cursor)
+                    # Only the last part can be smaller than ``S3_MPU_PART_MAX_SIZE``.
+                    chunk_size = min(S3_MPU_PART_MAX_SIZE, total_file_size - cursor)
+                    wrapped_object = _CustomCallbackIOWrapper(ctx.update, f, 'read', chunk_size)
+                    s = Session()
+                    req = Request('PUT', url_info['upload_url'], data=wrapped_object)
+                    prep = req.prepare()
+                    prep.headers['Content-Length'] = chunk_size
+                    response = s.send(prep)
+                    if response.status_code != 200:
+                        secho_error_and_exit(f"Failed to upload file ({file_path}): {response.content}")
```

Comment:
maybe we have to just call `reponse.raise_for_status`?
Exit directly at here would not call `abort_mpu`